src/plugins/discover/public/application/main/utils/get_valid_view_mode.ts

```diff
@@ -20,10 +20,5 @@ export const getValidViewMode = ({
   viewMode?: VIEW_MODE;
   isTextBasedQueryMode: boolean;
 }): VIEW_MODE | undefined => {
-  if (viewMode === VIEW_MODE.AGGREGATED_LEVEL && isTextBasedQueryMode) {
-    // only this mode is supported for text-based languages
-    return VIEW_MODE.DOCUMENT_LEVEL;
-  }
-
```

Comment:
@stratoula Do we want to enable now adding ES|QL field stats as a panel to dashboards? 

In this PR it resets `viewMode` to `documents` still when saving a search. Maybe because of https://github.com/elastic/kibana/blob/828b5c9da4c6c17b119d6e1c2048254e1b4cb779/src/plugins/discover/public/application/main/state_management/utils/build_state_subscribe.ts#L55

Reply:
Does this line enable that? Hmm, is it possible on dataview mode?

Reply:
I see, I didn't know that we have this functionality 😄 It is not very obvious that when you save a saved search in the field statistics tab, you embed this view instead of the table. 🤔 

My personal opinion is for now to always embed the datatable and not store the selected tab. I will add it as a discussion point at the next ES|QL meeting to decide the next steps. Maybe we can improve the experience here.

Reply:
I just tested it again in this PR and looks like it got changed to show Field Statistics as a panel if the tab was active when saving in ES|QL mode too. I don't mind it either way but would be great to agree now how it should work.

Reply:
I find it a very weird behavior tbh. I have added this as a conversation item for next week but I dont want to block this PR. Can we disable this behavior only for ESQL for now and create a follow up PR if necessary to enable it after the bi-weekly?